device/samples/pnp/pnpTemperatureController.js

```diff
@@ -0,0 +1,256 @@
+        
+        patch[componentName][propertyName] = propertyContent;
+    }});
+    updateComponentReportedProperties(deviceTwin, patch, componentName);
+ }});
+};
+
+const helperAttachExitListener = async (deviceClient) => {
+  const standardInput = process.stdin;
+  standardInput.setEncoding('utf-8');
+  console.log('Please enter q or Q to exit sample.');
+  standardInput.on('data', (data) => {
+    if (data === 'q\n' || data === 'Q\n') {
+      console.log('Clearing intervals and exiting sample.');
+      clearInterval(intervalToken1);
+      clearInterval(intervalToken2);
+      clearInterval(intervalToken3);
+      deviceClient.close();
+      process.exit();
+    } else {
+      console.log('User Input was : ' + data);
+      console.log('Please only enter q or Q to exit sample.');
+    }
+  });
+};
+async function sendTemperatureTelemetry(deviceClient, componentName, index) {
+  console.log('Sending telemetry message %d from component %s ', index, componentName);
+  
+  const data = JSON.stringify({ temperature: 1 + (Math.random() * 90)});
+  const pnpMsg = new Message(data);
+  if (!!(componentName))
+  {
+    pnpMsg.properties.add(messageSubjectProperty, componentName);
+  }
+  pnpMsg.contentType = 'application/json';
+  pnpMsg.contentEncoding = 'utf-8';
+  await deviceClient.sendEvent(pnpMsg);
+}
+
+async function sendDatasizeTelemetry(deviceClient, componentName, index) {
+  console.log('Sending telemetry message %d from root interface', index);
+  const data = JSON.stringify({ workingset: 1 + (Math.random() * 90)});
+  const pnpMsg = new Message(data);
+  if (!!(componentName))
+  {
+    pnpMsg.properties.add(messageSubjectProperty, componentName);
+  }
+  pnpMsg.contentType = 'application/json';
+  pnpMsg.contentEncoding = 'utf-8';
+  await deviceClient.sendEvent(pnpMsg);
+}
+
+async function main() {
+  // fromConnectionString must specify a transport, coming from any transport package.
+  const client = Client.fromConnectionString(deviceConnectionString, Protocol);
+  console.log('Connecting using connection string ' + deviceConnectionString);
+  let resultTwin;
+
+  try {
+    // Add the modelId here
+    await client.setOptions({ modelId: modelId });
+    await client.open();
+    console.log('Enabling the commands on the client');
+    client.onDeviceMethod(commandNameGetMaxMinReport1, commandHandler);
+    client.onDeviceMethod(commandNameGetMaxMinReport2, commandHandler);
+    client.onDeviceMethod(commandNameReboot, commandHandler);
+
+    // Send Telemetry after some interval
+    let index1 = 0;
+    let index2 = 0;
+    let index3 = 0;
+    intervalToken1 = setInterval(() => {
+      sendTemperatureTelemetry(client, thermostat1ComponentName, index1).catch((err) => console.log('error', err.toString()));
+      index1 += 1;
+    }, 5000);
+
+    intervalToken2 = setInterval(() => {
+      sendTemperatureTelemetry(client, thermostat2ComponentName, index2).catch((err) => console.log('error', err.toString()));
```

Comment:
>r' [](start = 105, length = 2)

Seems as though there should be a space in the error string.   Perhaps 'error: '   ?

Reply:
spaced now..but u said colons are not used in node js logs...error is just a higher level of log right ?